language/Type-Classes.md

```diff
@@ -80,6 +80,69 @@ In fact, a type similar to this `AddInt` is provided in `Data.Monoid.Additive`,
 
 For multi-parameter type classes, the orphan instance check requires that the instance is either in the same module as the class, or the same module as at least one of the types occurring in the instance. (TODO: example)
 
+## Kinds
+
+A typeclass's type parameter can specify the kind of types it applies to. If it doesn't specify the type's kind, it can be inferred by the typeclass's methods.
```

Comment:
That's a lot more clear. I'll add that.

I was also curious about "typeclass" and "type class". I'll update this document to use "type class" if @hdgarrood agrees.